geth/peers/topicpool.go

```diff
@@ -169,7 +176,11 @@ func (t *TopicPool) StartSearch(server *p2p.Server) error {
 }
 
 func (t *TopicPool) handleFoundPeers(server *p2p.Server, found <-chan *discv5.Node, lookup <-chan bool) {
-	t.period <- t.fastSync
+	if t.connected >= t.limits[0] {
```

Comment:
This also seems like a possible thing to be refactored. `handleFoundPeers` does not only handle peers but also decides about fast vs slow syncing. It seems like too many responsibilities.

Reply:
what do you mean by refactoring here? i just need to decide on sync type before loop is started

Reply:
How does `t.period` influence the loop behaviour?